### DISCUSSION TITLE
Add babashka to sdkman.io package list

### DISCUSSION THREAD

**Discussion by rajcspsg** [babashka/babashka#1527] [Ideas]
**Is your feature request related to a problem? Please describe.**
 Lot of JVM related SDKs can be installed via sdman and sdkman is getting popular nowadays. Add this amazing package to sdkman.io package list. 

**Describe the solution you'd like**
The solution is easy way. We have to follow the vendor onboarding process [here](https://github.com/sdkman/sdkman-cli/wiki/Vendor-onboarding-process)

**Describe alternatives you've considered**
Brew is not allowed in our organization. I would be nice to have multiple ways to have an SDK installed.

**Comment by borkdude** [comment]
Package managers are usually maintained by third parties. I only maintain the brew package and this is enough work as it is but if someone wants to add sdkman support, go for it!

**Comment by marc0der** [comment]
> Package managers are usually maintained by third parties.

SDKMAN isn't a package manager but more of a download manager. Many of the organisations in the JVM space use our Vendor API to publish their new versions for download these days. You can read more about that on our [website](https://sdkman.io/vendors). You can also take a look at the SDKs we [already support](https://sdkman.io/seks).

If you wire JReleaser into your build, you get publishing to SDKMAN for free, all you need is an access token.

We're completely open source and don't make profit out of the tool. It would be great to have you join us.

**Reply by borkdude** [reply]
cc @lispyclouds: perhaps something we could look into in the coming months

**Reply by aalmiray** [reply]
JReleaser author here 👋 
You may use the tool with [GitHub Actions](https://jreleaser.org/guide/latest/continuous-integration/github-actions.html) or [CircleCI](https://jreleaser.org/guide/latest/continuous-integration/circle-ci.html) depending on how you-d like to build and release binaries.

Happy to help with the setup. We can always be reached at https://github.com/jreleaser/jreleaser/discussions

**Reply by borkdude** [reply]
@aalmiray babashka already uploads the binaries to Github Releases. I use a couple different CI systems: CircleCI, CirrusCI and Appveyor. All these systems simultaneously upload the binary to Github releases in a Draft release, creating it when necessary, using https://github.com/borkdude/gh-release-artifact. When everything ran successfully, I then manually toggle the draft release to a public release.

Can you tell me where JReleaser is necessary in this part or can I simply write a babashka script that pushes the latest Github release artifacts to sdkman? Or if SDKMAn is a Download manager, why doesn't it just download the binaries from Github releases, since that's what the canonical place for dowloading binaries for babashka is. Perhaps I'm missing something.

**Reply by aalmiray** [reply]
Well, JReleaser performs all those tasks with a single configuration file, both locally or on any CI. This means its behavior is reproducible regardless of the environment that is, no need to configure different GH actions for dealing with creating releases, uploading assets, generating changelog, etc. JReleaser can do it all if that’s what you may be looking for. Or just a few parts. There’s no need to overhaul/replace the whole release process with the tool. You can use as little or as much of it as needed. 

Now, regarding Sdkman. Sure, you can use the REST API, or the Maven/Gradle plugins, or JReleaser. The advantage of the latter is the reuse of common metadata (model configuration) in case you’d like to publish binaries to additional channels such as brew and scoop. Once you have the model publishing to brew requires 2 additional config lines. Same goes for scoop. You can do this with REST APIs and template files but that requires more work on your side and duplicate efforts/files.

**Reply by borkdude** [reply]
I'm not interested right now in rewriting any of the infrastructure that is in place and has worked well for years. 
What I like about the current approach is that I have a final moment in which I can say: now the Github release looks fine, let's make it public. When using JReleaser, I think I won't have that level of control anymore.

Let me just re-phrase stuff. I'm not interested in putting any effort in changing release infrastructure since I'm happy with how stuff currently works. But there are people who would like to download babashka with sdkman instead of Github actions. Given that:
What is the _minimal_ effort that I or someone else can do to satisfy sdkman users, given that the binaries are already on github releases?

**Reply by aalmiray** [reply]
Understood. In that case, a direct REST call to Sdkman's API would be enough. Doesn't require any external tools other than bash but does require knowledge of the given endpoints and payload format. 

OTOH, JReleaser requires 1 extra tool download (if running locally) or 1 extra GH action (if running on CI), plus 1 extra model file. With just 1 command (`jreleaser publish`) assuming binaries are already found at the expected location (tagged Github release).

**Reply by borkdude** [reply]
Thanks a lot. So I guess jreleaser could be used in an additional CI job that runs with a cron expression or so or is triggered otherwise, that would upload the Github release binaries to sdkman. Sounds good.

**Reply by aalmiray** [reply]
Correct. There’s one extra bit of information. I think right now the model expects binaries to be resolvable from the local filesystem before publication/announcing to sdkman can happen. This means downloading assets to a given location is a requirement. Luckily JReleaser has a `download` command that can do this but does come with a downside: “duplicate” asset defs. 1 url for download, 1 path for resolving the file for sdkman publication.

**Reply by lispyclouds** [reply]
I can try to take a stab as this over the weekend.
